cmd/devp2p/crawlercmd.go

```diff
@@ -0,0 +1,300 @@
+	}
+	defer conn.Close()
+	conn.SetDeadline(time.Now().Add(5 * time.Second))
+
+	// write hello to client
+	pub0 := crypto.FromECDSAPub(&sk.PublicKey)[1:]
+	ourHandshake := &ethtest.Hello{
+		Version: 5,
+		Caps: []p2p.Cap{
+			{Name: "eth", Version: 64},
+			{Name: "eth", Version: 65},
+			{Name: "eth", Version: 66},
+		},
+		ID: pub0,
+	}
+	if err := conn.Write(ourHandshake); err != nil {
+		return nil, err
+	}
+
+	// read hello from client
+	switch msg := conn.Read().(type) {
+	case *ethtest.Hello:
+		// set snappy if version is at least 5
+		if msg.Version >= 5 {
+			conn.SetSnappy(true)
+		}
+		info.Capabilities = msg.Caps
+		info.SoftwareVersion = msg.Version
+		info.ClientType = msg.Name
+		fmt.Printf("Caps: %v \n", msg.Caps)
+	case *ethtest.Disconnect:
+		return nil, fmt.Errorf("bad hello handshake: %v", msg.Reason.Error())
+	case *ethtest.Error:
+		return nil, fmt.Errorf("bad hello handshake: %v", msg.Error())
+	default:
+		return nil, fmt.Errorf("bad hello handshake: %v", msg.Code())
+	}
+	highestEthVersion := uint32(negotiateEthProtocol(ourHandshake.Caps, info.Capabilities))
+	// If node provides no eth version, we can skip it.
+	if highestEthVersion == 0 {
+		return &info, nil
+	}
+	conn.SetDeadline(time.Now().Add(15 * time.Second))
+	// write status message
+	status, err := getCurrentStatus()
+	if err != nil {
+		return nil, err
+	}
+	status.ProtocolVersion = highestEthVersion
+	if err := conn.Write(status); err != nil {
+		return nil, err
+	}
+
+	// read status message from client
+	switch msg := conn.Read().(type) {
+	case *ethtest.Status:
+		info.ForkID = msg.ForkID
+		info.HeadHash = msg.Head
+		info.NetworkID = msg.NetworkID
+		// m.ProtocolVersion
+		info.TotalDifficulty = msg.TD
+	case *ethtest.Disconnect:
+		return nil, fmt.Errorf("bad status handshake: %v", msg.Reason.Error())
+	case *ethtest.Error:
+		return nil, fmt.Errorf("bad status handshake: %v", msg.Error())
+	default:
+		return nil, fmt.Errorf("bad status handshake: %v", msg.Code())
+	}
+
+	// Disconnect from client
+	conn.Write(ethtest.Disconnect{Reason: p2p.DiscQuitting})
+
+	return &info, nil
+}
+
+// dial attempts to dial the given node and perform a handshake,
+func dial(n *enode.Node) (*ethtest.Conn, *ecdsa.PrivateKey, error) {
+	var conn ethtest.Conn
+	// dial
+	fd, err := net.Dial("tcp", fmt.Sprintf("%v:%d", n.IP(), n.TCP()))
+	if err != nil {
+		return nil, nil, err
+	}
+	conn.Conn = rlpx.NewConn(fd, n.Pubkey())
+	// do encHandshake
+	ourKey, _ := crypto.GenerateKey()
+	_, err = conn.Handshake(ourKey)
+	if err != nil {
+		return nil, nil, err
+	}
+	return &conn, ourKey, nil
+}
+
+func getCurrentStatus() (*ethtest.Status, error) {
+	raw, err := rpc.Dial("wss://mainnet.infura.io/ws/v3/2e5a3920f039435d9bb23729c7b65186")
+	if err != nil {
+		return nil, err
+	}
+	cl := ethclient.NewClient(raw)
+
+	nwid, err := cl.NetworkID(context.Background())
+	if err != nil {
+		return nil, err
+	}
+
+	genesis, err := cl.HeaderByNumber(context.Background(), big.NewInt(0))
```

Comment:
You're using mainnetchainconfig as forkid. So you already knows the genesis hash at compile time. 

If you want the crawler to be configurable, you could make it so the user can either specify the network (`--ropsten`, `--rinkeby` etc), and set the genesis hash  programmatically, and you can also make it totally configurable, `--genesishash 0x...`)

Seems unwarranted to query infura about it